### PR TITLE
Feat/pablo/update server users

### PR DESCRIPTION
**Pull request by pabloredsky**
Added logic in order to save users to server. Validation added adding a superForm form.

### CODE REVIEW

File: src/lib/server/simpleDb.ts [hunk 1]
```diff
@@ -84,22 +84,39 @@ class SimpleDb {
...
+		return true;
+	}
+
+	async writeData(fileName: string, data: object) {
 		try {
+			console.log(typeof data);
 			await fs.writeFile(
-				STORAGE_PATH + '/users.json',
-				JSON.stringify(this.users, null, 2)
+				`${STORAGE_PATH}/${fileName}.json`,
```
Comment (JoshuaHintze) [inline src/lib/server/simpleDb.ts]: anywhere where you use STORAGE_PATH we should use path.join(). Eventhough this is a built in node package I think we can use it.

File: src/lib/ui/serverInfoCard/ServerInfoCard.svelte [hunk 2]
```diff
@@ -15,25 +15,44 @@
...
 
+	let { serverInfo, users, onNextServer, onPreviousServer, addUsersToServer }: Props = $props();
 	let isAddToUsersDialogOpen = $state(false);
 	let userIdsToAdd = $state<number[]>([]);
 
 	const usersAvailableToAdd = $derived(users.filter((user) => !serverInfo.userIds.find((id) => id === user.id)));
+	const addUsersToServerForm = superForm(addUsersToServer, {
+		validators: zod(addUsersToServerSchema),
+		onUpdated: ({ form }) => {
+			console.log('form', form);
```
Comment (JoshuaHintze) [inline src/lib/ui/serverInfoCard/ServerInfoCard.svelte]: remove console.log

File: src/routes/(auth)/dashboard/+page.server.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { addUsersToServerSchema } from '$lib/schema/schema';
+
+export const actions: Actions = {
+	// Other actions...
+
+	'add-users-to-server': async ({ request }) => {
+		const form = await superValidate(request, zod(addUsersToServerSchema));
+		
+		if (!form.valid) return fail(400, { form });
+		console.log('form.data', form.data);
```
Comment (JoshuaHintze) [inline src/routes/(auth)/dashboard/+page.server.ts]: remove console log

**Review by JoshuaHintze — CHANGES_REQUESTED** [review]

File: src/lib/server/simpleDb.ts (hunk 1 above)
Comment (pabloredsky) [reply src/lib/server/simpleDb.ts]: Alright

File: src/lib/ui/serverInfoCard/ServerInfoCard.svelte (hunk 2 above)
Comment (pabloredsky) [reply src/lib/ui/serverInfoCard/ServerInfoCard.svelte]: Okay

File: src/routes/(auth)/dashboard/+page.server.ts (hunk 3 above)
Comment (pabloredsky) [reply src/routes/(auth)/dashboard/+page.server.ts]: Okay, i added this because i saw you were doing the same in the other actions

Comment (JoshuaHintze) [reply src/routes/(auth)/dashboard/+page.server.ts]: My bad. Yeah, but its a good habbit to not get into.

**Review by JoshuaHintze — APPROVED** [review]

**Comment by pabloredsky** [comment]
Fixes #2

**Pull request merged**